### PR TITLE
Implement string date function

### PR DESCRIPTION
**Pull request by caldwellst**
Reviewing your GLOFAS module inspired me to tackle some of the backlog in particular #128 and #129.

### CODE REVIEW

**Review by turnerm — COMMENTED** [review]
This is so great Seth, thanks for implementing it! Just one small bug to fix. 

It _did_ make me notice a couple of additional things that would also be very quick fixes and great to throw in, if you wouldn't mind (sorry):

- For NDVI, there are no logger messages for downloading / processing, would you be able to add them? (e.g., {year} and {dekad} is being downloaded to {filepath}). If you're worried about logger spam you can make it `DEBUG`. 
- In the `_Chirps` class I noticed that there are a lot of start dates / end dates being passed around where I think the instance variables can just be used -- if I'm not wrong, would you mind changing the methods to use instance variables? Thanks so much. (Unless @castledan there is a reason why you didn't do it this way?)

**Comment by caldwellst** [comment]
Thanks Monica! Have added in the logger messages to NDVI and used instance variables across CHIRPS (all tests still passing and reviewing the code I saw no reason not to), as well as addressed the other comments!

**Comment by castledan** [comment]
Thank you Seth for fixing the CHIRPS code. I'm not sure why I'd implemented it in a different way, I don't think there was a particular reason.

File: src/aatoolbox/datasources/usgs/ndvi_base.py [hunk 1]
```diff
@@ -487,6 +488,11 @@ def _download(self, filepath: Path, url_filename: str, clobber: bool):
                 )
                 return filepath
 
+        logger.info(
+            f"Downloading NDVI data for {year}, dekad {dekad} "
+            "into {filepath}."
```
Comment (turnerm) [inline src/aatoolbox/datasources/usgs/ndvi_base.py:493]: Missing the `f` here

**Review by turnerm — APPROVED** [review]
Looks good to me, go ahead and merge (just one typo to fix). I will merge this into my own GloFAS branch.

**Pull request merged**